### PR TITLE
screengrab screenshot strategies

### PR DESCRIPTION
**Pull request by mfurtak**
Refactors `Screengrab` into more collaborators, while keeping the existing `Screengrab` interface working.

The goal is to introduce a new strategy for screenshot capture based on [UiAutomation](https://github.com/fastlane/fastlane/issues/2080#issuecomment-193459342). It allows capture on Android N, and more correctly captures multi-layer/window situations (dialogs, etc.) Some verification is needed to determine if it also helps Google Maps & video player use-cases.

The new strategy depends on `UiAutomation` which has an API >= 18 requirement. We enforce that in code, and override the manifest merger checks that would force the (optional) API level requirement onto consuming apps. So, our `minSdkVersion` remains **8** for the library.

The `JUnit3StyleTests` are removed, as they are not a recommended configuration.

The example application and tests have been slightly expanded to include a button that opens a dialog to verify multi-window capture by `UiAutomatorScreenshotStrategy`
#### To Do:
- [x] Document new classes and interfaces
- [x] Update README documentation to describe using the strategies

### CODE REVIEW

File: screengrab/lib/screengrab/runner.rb [hunk 1]
```diff
@@ -220,7 +220,7 @@ def run_tests(device_serial, test_classes_to_use, test_packages_to_use)
                                         print_all: true,
                                         print_command: true)
 
-        UI.user_error! "Tests failed" if test_output.include?("FAILURES!!!")
+        UI.user_error!("Tests failed", show_github_issues: false) if test_output.include?("FAILURES!!!")
```
Comment (asfalcone) [inline screengrab/lib/screengrab/runner.rb]: NIce!

**Comment by asfalcone** [comment]
This is looking great! Awesome refactoring 💯

**Comment by mfurtak** [comment]
Thanks! Ready for a final review. 👍

File: screengrab/screengrab-lib/src/main/java/tools.fastlane.screengrab/Screengrab.java [hunk 2]
```diff
@@ -24,37 +24,72 @@
...
+     *                 the default ScreenshotStrategy for this invocation
+     * @see #setDefaultScreenshotStrategy(ScreenshotStrategy)
+     */
     public static void screenshot(String screenshotName, ScreenshotStrategy strategy) {
         Context appContext = InstrumentationRegistry.getInstrumentation()
                 .getTargetContext()
                 .getApplicationContext();
 
         screenshot(screenshotName, strategy, new FileWritingScreenshotCallback(appContext));
+        new HashMap();
```
Comment (asfalcone) [inline screengrab/screengrab-lib/src/main/java/tools.fastlane.screengrab/Screengrab.java]: Is this dangling `HashMap()` out of place?

**Comment by asfalcone** [comment]
Looks 💯 Just the one comment about a rando `new HashMap()` 👍

File: screengrab/screengrab-lib/src/main/java/tools.fastlane.screengrab/Screengrab.java (hunk 2 above)
Comment (mfurtak) [inline screengrab/screengrab-lib/src/main/java/tools.fastlane.screengrab/Screengrab.java]: Yeah, that shouldn't have snuck in, thanks!

**Comment by asfalcone** [comment]
:shipit:

**Pull request merged**